src/main/java/org/jenkinsci/plugins/workflow/steps/StepDescriptor.java

```diff
@@ -305,6 +306,27 @@ public boolean apply(StepDescriptor i) {
         return r;
     }
 
+    /** Return true if we can easily create a nice String for user display from the object */
+    static boolean isAbleToUseToStringForDisplay(@CheckForNull Object o) {
+        return (o != null && (o instanceof Comparable || o instanceof CharSequence)); // Covers our base types
+    }
+
+    /**
+     * Converts user-supplied step arguments to a string for eventual UI use -- override to handle more than a single trivial argument.
+     * Complements {@link #getDisplayName()} in cases where the step type is less meaningful than its arguments (scripts, for example).
+     * <em>Note: this offers a raw value and does not perform escaping on its own.</em>
+     * @param namedArgs List of parameter name-value pairs supplied to the step to instantiate it, as from {@link #uninstantiate(Step)}
+     * @return Formatted string, before escaping, or null if can't be converted easily to a String.
+     */
+    @Nullable
```

Comment:
`@Nullable` is the default for this package. Given that `null` is an expected outcome it may be better to either use `@CheckForNull` or omit the annotation.

Most people get confused by `@Nullable` saying that 

> The annotated element could be `null` under some circumstances. In general, this means developers will have to read the documentation to determine when a `null` value is acceptable and whether it is necessary to check for a `null` value.

The documentation you provide does not define the full set of criteria when `null` will be returned, e.g. doesn't cover the case where the single argument value is `null` nor the case where there are multiple arguments nor the case where the supplied map is `null`... and then overridden implementations may have entirely different criteria under which the returned value will be `null`... and I cannot read all implementations.

In other words (and this is why I am negging on this specific use of an annotation) this method is *both* an API and an SPI, thus you either should document strictly the cases for returning `null` so that implementers have a strong contract to follow (i.e. make the SPI strict enough that the documentation is complete in the method itself) or switch to an API/SPI split (which is not the existing pattern in this class and will likely cause confusion)... or maybe switch to `@CheckForNull`